cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ComputeEngineCredentials.java

```diff
@@ -123,9 +130,40 @@ private AccessToken refreshAccessTokenRemotely(String endPoint) throws IOExcepti
     throw new IOException(exception.getMessage(), exception);
   }
 
+  private void disableVerifySSL(HttpsURLConnection connection) throws IOException {
+    try {
+      SSLContext sslContext = SSLContext.getInstance("SSL");
+      TrustManager[] trustAllCerts = new TrustManager[]{new X509TrustManager() {
+        public X509Certificate[] getAcceptedIssuers() {
+          return null;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+          // No-op
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+          // No-op
+        }
+      }};
+      sslContext.init(null, trustAllCerts, new SecureRandom());
```

Comment:
Don't create SecureRandom each time, it's expensive. Move it to constructor, probably along with SSLContext.getInstance

Reply:
Done.